### PR TITLE
Fix LinearColorSmoothing CPU usage

### PR DESCRIPTION
**Pull request by vexofp**
Fixed LinearColorSmoothing in continuousOutput=false mode to not consume CPU for no reason after the output has settled.

### CODE REVIEW

File: libsrc/hyperion/LinearColorSmoothing.cpp [hunk 1]
```diff
@@ -40,30 +40,54 @@ int LinearColorSmoothing::write(const std::vector<ColorRgb> &ledValues)
...
+	}
+	else
 	{
-		_outputQueue.push_back(_targetValues);
-		_outputQueue.pop_front();
+		_ledDevice->switchOff();
+
+		if(_writeToLedsEnable)
+		{
+			_timer.stop();
```
[the remark this answers is not in the document]
Comment (vexofp) [reply libsrc/hyperion/LinearColorSmoothing.cpp]: With these changes, `_timer` and `_writeToLedsEnable` are always in one of two states:
* `_writeToLedsEnable == false` and `_timer` is stopped (this is the initial state)
* `_writeToLedsEnable == true` and `_timer` is running

Thus, the timer only needs stopping when `_writeToLedsEnable` is set, and vice versa.

`_writeToLedsEnable` is set to `true` (and `_timer` started if not already running) any time we need to start sending pixel data to the LEDs. This happens when `write()` is called, or when `switchOff()` is called but we are in `_continuousOutput` mode and thus need to send black pixels.

For `switchOff()` in non-continuous mode, we invoke the device's `switchOff()` method once and then stop sending pixel data until/unless `write()` is called again.

File: libsrc/hyperion/LinearColorSmoothing.cpp [hunk 2]
```diff
@@ -114,14 +142,12 @@ void LinearColorSmoothing::queueColors(const std::vector<ColorRgb> & ledColors)
 	if (_outputDelay == 0)
 	{
 		// No output delay => immediate write
-		if ( _writeToLedsEnable )
```
[the remark this answers is not in the document]
Comment (vexofp) [reply libsrc/hyperion/LinearColorSmoothing.cpp]: `updateLeds()` (and thus `queueColors()`) is called if and only if `_timer` is running. As discussed in the previous comment, `_timer` will be now running if and only if `_writeToLedsEnable` is set. Thus, these checks are now redundant.

File: libsrc/hyperion/LinearColorSmoothing.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (WoLpH) [reply libsrc/hyperion/LinearColorSmoothing.cpp]: That makes sense :)

**Comment by WoLpH** [comment]
Thanks for the fix!

**Pull request merged**